{% extends 'plots/base.html' %}

{% block content %}

{% load static %}

<div class="w3-row w3-padding header1-dark">
  <div class="w3-half w3-container">
    <h4 class="header1-dark"><b>THE DATASET</b></h4>
    <p class="w3-large paragraph-d" style="text-align:justify">Single cell gene expression data was obtained from a study in which human megakaryocyte-erythroid progenitor (MEP) cells were isolated and a panel of 87 genes profiled to better understand the differentiation process of erythrocytes and megakaryocytes in human hematopoiesis (REF TO PAPER HERE). MEP cells are able to differentiate into either erythrocytes or megakaryocytes and form part of the hematopoietic stem cell differentiation process laid out in Figure 4 below.</p>
    <img class="img-center" src="{% static 'plots/images/workflow1.png' %}" style="width: 100%">
    <figcaption class="paragraph-d "><b>Figure 4: </b>The traditional hierarchical model of HSC stem cell differentiation with MEP cells bifurcating to become lineage-commited erythroids or megakaryocytes.</figcaption>
    <p class="w3-large paragraph-d" style="text-align:justify">CD34+ blood cells were obtained from three healthy volunteers, cells were stained using a 10-fluorochrome panel and MEP cells sorted using flow cytometry. Individual cells were aliquoted into 96-well PCR plates for quantitative real-time PCR. An overview of the experimental protocol used to generate the data is outlined in Figure 5.</p>
  </div>

  <div class="w3-half w3-container">
    <img class="figwork" src="{% static 'plots/images/workflow.png' %}">
    <figcaption class="figwork-pad w3-center paragraph-d"><b>Figure 5: </b>Experimental workflow followed by Psaila <i>et al.</i> to obtained MEP cells from whole blood.</figcaption>
  </div>
</div>
<br>


<div class="w3-row w3-padding header1-light">
  <div class="w3-half w3-container">
    <!-- I want to split this in two horizontally and have 2 images in black and white -->
    <h4 class="invisible"><b>FEATURE EXTRACTION & MUTUAL INFORMATION</b></h4>
    <p class="w3-large paragraph-l" style="text-align:justify">While relying on statistical tools to determine which pairs of genes may be most informative is useful, it is also important to be guided by the biological process in question and the literature available on which genetic players may drive and guide these processes. Consequently, it is important to have the ability to compute the landscape for any possible pair of genes and not only pairs characterised by a high mutual information. All possible pairs of genes within the dataset and the mutual information between them is presented in the form of a heatmap and is presented under the data landscape section of the website.
    </p>
    <img class="figmi" src="{% static 'plots/images/mi.png' %}">
    <figcaption class="figmi-pad w3-center paragraph-l "><b>Figure 6: </b>A heatmap representing all possible pairs of genes from the dataset and the corresponding mutual inforamtion each pair shares on a scale from 0 to 1.</figcaption>
  </div>

  <div class="w3-half w3-container">
    <h4 class="header1-light"><b>FEATURE EXTRACTION & MUTUAL INFORMATION</b></h4>
    <p class="w3-large paragraph-l" style="text-align:justify">Since epigenetic landscapes are 3D surface plots it is only possible to render a landscape for a pair of genes with the Z-coordinate encapsulating the potential landscape (see below section for further details). Consequently, feature extraction is necessary to determine which pairs of genes might yield the most informative landscapes. While Pearson Correlation is often used to measure the linear relationship between two variables a major downfall is it’s inability to capture non-linear dependencies in the data. To that end, tools from information theory are employed such as mutual information. Mutual information measures the relationship between two random variables that are sampled simultaneously. In simple terms, we may think of it as how much one random variable tells us about the other or how much mutual information they share.The mutual information between two random variables `X` and `Y`, whose joint distribution is defined by `P(X,Y)` and where `p(x)` and `p(y)` are the marginal probability distribution functions of `X` and `Y`, respectively is given by:</p>
    <p class="w3-large paragraph-l w3-center">`I(X;Y)=sum_(x in X)sum_(yinY)p(x,y)log(p(x,y)/p(x)p(y))`</p>
    <!-- <p class="w3-large paragraph-l" style="text-align:justify">Where: `p(x)` and `p(y)` are the marginal probability distribution functions of `X` and `Y`, respectively.</p> -->
    <p class="w3-large paragraph-l" style="text-align:justify">The mutual information for each possible pair of genes is calculated using the Julia package InformationMeasures.jl and data added to the backend of the website.</p>
  </div>
</div>
<br>


<div class="w3-row w3-padding header1-dark">
  <div class="w3-half w3-container">
    <h4 class="header1-dark"><b>GENERATING THE LANDSCAPES</b></h4>
    <p class="w3-large paragraph-d" style="text-align:justify">While the Waddington landscape was only ever meant to be a qualitative representation of how gene expression directs cellular differentiation, mathematicians have in recent years put forward several proposals to construct a quantitative energy landscape. Different proposals are comprehensively reviewed in XXX including the limitations of the energy landscape proposal, but in essence the potential landscape is computed by obtaining the negative logarithm of the steady-state probability distribution. Formally:</p>
    <p class="w3-large paragraph-d w3-center">`U = -logPs`</p>
    <p class="w3-large paragraph-d" style="text-align:justify">Where `Ps` is the steady state of the probability distribution and U represents the potential landscape of the system.</p>
    <p class="w3-large paragraph-d" style="text-align:justify">In this project the probability distribution is estimated using kernel density estimation – a non-parametric technique used to estimated the probability distribution of a random variable. Computations are implemented using the Python Gaussian kernel density estimation function scipy.stats.gaussian_kde.
    </p>
    <p class="w3-large paragraph-d" style="text-align:justify">Due to the vast number of potential landscapes that can be generated given the dataset, combined with a need to analyse any landscape the user sees fit (those with high mutual information or those which don’t share a high mutual information, but are biologically interesting) a website backend containing a dynamic relational database is imperative. While it is possible to create 1500 static landscapes and load the images into the Django environment, it is far more practical and dynamic to compute and generate the landscapes in real time making use of Django’s Python framework. Although this requires a robust backend to be setup with initial labour costs, it allows significant downstream flexibility for both the creator and the user.</p>
    <p class="w3-large paragraph-d" style="text-align:justify">In order to plot a 3D landscape the gene expression values for gene 1 and gene 2 are required to compute the potential landscape or the steady-state probability distribution (Z-dimension of the plot). While the expression values for each gene are contained within the database of the website, the probability distribution for a given X and Y is computed in real-time in the Django views, depending on which pair of genes the user would like to visualise the landscape for. The potential landscape is subsequently stored as a Django variable ready to be deployed into a JavaScript environment for visualisation.</p>
  </div>

  <div class="w3-half w3-container">
    <h4 class="header1-dark"><b>VISUALIZING THE LANDSCAPES</b></h4>
    <p class="w3-large paragraph-d" style="text-align:justify">Landscapes are plotted using the plotly.js graphing library allowing the user the ability to fully explore the landscape due to the interactivity associated with JavaScript. The X, Y and Z co-ordinates for any chosen landscape are imported from the views to the JavaScript graphing code to produce the user’s landscape of choice, in both 2 and 3 dimensions. A colour bar indicates the height of the Z dimension/potential landscape with the lightest colours representing the crests and the darkest colours the valleys, corresponding to basins of attraction where cells are most likely to end up. The landscapes are accessed from the data landscape tab where a user is first presented with a heatmap representing mutual information between all gene pairs. The user is then able to click on any block within the heatmap (representing a pair of genes), thereby generating the landscape in real time.</p>
    <img class="figland" src="{% static 'plots/images/new_landscape.png' %}" style="width=100%">
    <figcaption class="figland-pad w3-center paragraph-d"><b>Figure 7: </b>Image of a 3D epigenetic landscape generated by this website with darkest colours corresponding to basins of attraction where cells are most likely to end up.</figcaption>
  </div>
</div>
<br>


<div class="w3-row w3-padding header1-light">
  <div class="w3-half w3-container">
    <img class="figmi" src="{% static 'plots/images/pca.png' %}">
    <figcaption class="figpca-pad w3-center paragraph-l"><b>Figure 8: </b>Principal component analysis shows that the MEP cell population is divided into two sub-groups by principal component 1.</figcaption>
  </div>

  <div class="w3-half w3-container">
    <h4 class="header1-light"><b>PRINCIPAL COMPONENT ANALYSIS WITH GENE EXPRESSION OVERLAY</b></h4>
    <p class="w3-large paragraph-l" style="text-align:justify">Principal component analysis (PCA) is a useful statistical tool that makes use of orthogonal transformation to reduce high dimensional datasets into a set of linearly uncorrelated variables or principal components.</p>
    <p class="w3-large paragraph-l" style="text-align:justify">To further explore the dataset PCA is performed and shows that MEP cells are divided into two sub-populations by principal component 1. To begin investigating whether these sub-populations may potentially be two distinct groups exhibiting differential gene activity, each individual cell is able to be coloured according to the expression intensity of a particular gene within that cell. This represents an intuitive and visually assertive way to identify whether certain genes are more highly or lowly expressed in these MEP sub-populations. Again, as this data is part of the website backend it allows the user to explore differential gene expression in the dataset in an interactive way and overlay the expression intensity of any gene within the dataset onto the MEP sub-populations as defined by PCA.</p>
    <p class="w3-large paragraph-l" style="text-align:justify">Click the <b class="w3-text-theme">EXPLORING HETEROGENEITY</b> icon below to start the interactive session. Alternatively, click the <b class="w3-text-theme">EXPLORING THE LANDSCAPE</b> icon to begin visualising the landscapes.</p>
  </div>
</div>
<br>

<div class="w3-row-padding w3-center w3-margin-top">
<div class="w3-half">
  <div class="w3-card w3-container" style="min-height:360px">
  <h4 class="header1-dark"><b>EXPLORING HETEROGENEITY</b></h4><br>
  <a href="/plots/hetero/?gene=1"> <i class="fa fa-line-chart w3-margin-bottom w3-text-theme" style="font-size:120px"></i></a>
  <p class="w3-large paragraph-d">Click the icon to explore the heterogeneity of the dataset</p>

  </div>
</div>

<div class="w3-half">
  <div class="w3-card w3-container" style="min-height:360px">
  <h4 class="header1-dark"><b>EXPLORING THE LANDSCAPE</b></h4><br>
  <a href="/plots/mi1/"> <i class="fa fa-bar-chart w3-margin-bottom w3-text-theme" style="font-size:120px"></i></a>
  <p class="w3-large paragraph-d">Click the icon to explore the 3D epigenetic landscapes</p>

  </div>
</div>
</div>
<br>

<!-- Footer -->
<footer class="w3-container w3-theme-d6 w3-padding-16 w3-center">
  <div class="w3-padding-32">
    <a href="/plots/hetero/?gene=1"> <button class="w3-btn w3-large button1 w3-hover-light-grey" onclick="document.getElementById('id01').style.display='block'" style="font-weight:400;">NEXT SECTION</button></a>
  </div>
  <ul>
    <li class="inline">MEGAN COOMER</li>
    <li class="inline">|</li>
    <li class="inline">IMPERIAL COLLEGE</li>
    <li class="inline">|</li>
    <li class="inline"><a href="https://www.w3schools.com/w3css/w3css_templates.asp">W3-SCHOOLS</a></li>
  </ul>
</footer>

{% endblock %}
